### ISSUE TITLE
Fail to create Service Fabric Managed Cluster: Error "networkSecurityRules.direction contains duplicate items"

### ISSUE THREAD

**Issue by pedrofv** [hashicorp/terraform-provider-azurerm#16206]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

2.96

### Affected Resource(s)/Data Source(s)

azurerm_service_fabric_managed_cluster

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 2.96.0"
    }
  }

  # Backend state is configured via deployment script
  backend "azurerm" {
        resource_group_name  = ""
        storage_account_name = ""
        container_name       = ""
        key                  = ""
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "mps_rg" {
  name     = "${var.partition_name}-mps-sf"
  location = var.partition_region

  tags = { "rg_scope" : "partition" }
}

resource "azurerm_service_fabric_managed_cluster" "mps-sf" {
  name                = "${var.partition_name}-mps-sf"
  
  resource_group_name = azurerm_resource_group.mps_rg.name
  location            = var.partition_region
  http_gateway_port   = 19080
  client_connection_port = 19000

  lb_rule {
    backend_port       = 20443
    frontend_port      = 443
    probe_protocol     = "https"
    probe_request_path = "/test"
    protocol           = "tcp"
  }
  lb_rule {
    backend_port       = 21443
    frontend_port      = 1443
    probe_protocol     = "https"
    probe_request_path = "/test"
    protocol           = "tcp"
  }
  
  username           = "admin"
  password           = "adminpwd"

  node_type {
    data_disk_size_gb      = 130
    name                   = "type0"
    primary                = true
    application_port_range = "20001-49000"
    ephemeral_port_range   = "10000-20000"

    vm_size            = "Standard_D2a_v4"
    vm_image_publisher = "MicrosoftWindowsServer"
    vm_image_sku       = "2019-Datacenter"
    vm_image_offer     = "WindowsServer"
    vm_image_version   = "latest"
    vm_instance_count  = 5
  }
  
  tags = { "rg_scope" : "partition" }
}







variable "partition_name" {
  type = string
}

variable "partition_region" {
  type = string
}

variable "tenant_id" {
  type = string
}
```

### Debug Output/Panic Output

```shell
azurerm_service_fabric_managed_cluster.mps-sf: Creating...
╷
│ Error: while creating cluster "test-mps-sf": managedcluster.ManagedClusterClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="Argument networkSecurityRules.priority and networkSecurityRules.direction contains duplicate items: 1000 and inbound"
│
│   with azurerm_service_fabric_managed_cluster.mps-sf,
│   on mps-sf.tf line 19, in resource "azurerm_service_fabric_managed_cluster" "mps-sf":
│   19: resource "azurerm_service_fabric_managed_cluster" "mps-sf" {
│
│ while creating cluster "test-mps-sf": managedcluster.ManagedClusterClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter"
│ Message="Argument networkSecurityRules.priority and networkSecurityRules.direction contains duplicate items: 1000 and inbound"
```

### Expected Behaviour

A new Service Fabric Managed instance should've been created.

### Actual Behaviour

Error during 'apply'

### Steps to Reproduce

1. terraform apply

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/service_fabric_managed_cluster#lb_rule

**Comment by sinbai** [comment]
@pedrofv thank you for opening this issue. I have submitted PR #16481 to fix this issue. Please track it to see whether you would then be unblocked when it is released.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against an early major version of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
